internal/cmd/run/cmd.go

```diff
@@ -1,165 +1,203 @@
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 package run
 
 import (
+	"errors"
 	"flag"
+	"fmt"
 	"os"
 	"runtime"
-	"time"
-
-	"github.com/spf13/cobra"
-	"github.com/spf13/pflag"
-	"k8s.io/client-go/tools/leaderelection/resourcelock"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/healthz"
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	zapl "sigs.k8s.io/controller-runtime/pkg/log/zap"
-	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+	"strings"
 
 	"github.com/operator-framework/helm-operator-plugins/internal/metrics"
 	"github.com/operator-framework/helm-operator-plugins/internal/version"
 	"github.com/operator-framework/helm-operator-plugins/pkg/annotation"
-	"github.com/operator-framework/helm-operator-plugins/pkg/manager"
+	"github.com/operator-framework/helm-operator-plugins/pkg/flags"
+	helmmgr "github.com/operator-framework/helm-operator-plugins/pkg/manager"
 	"github.com/operator-framework/helm-operator-plugins/pkg/reconciler"
 	"github.com/operator-framework/helm-operator-plugins/pkg/watches"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	zapf "sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
 )
 
+var log = logf.Log.WithName("cmd")
+
+// TODO: Print the helm-operator plugin version. Earlier this was
+// tied to operator-sdk version
+func printVersion() {
+	log.Info("Version",
+		"Go Version", runtime.Version(),
+		"GOOS", runtime.GOOS,
+		"GOARCH", runtime.GOARCH,
+		"helm-operator", version.GitVersion,
+		"commit", version.GitCommit)
+}
+
 func NewCmd() *cobra.Command {
-	r := run{}
+	f := &flags.Flags{}
```

Comment:
The idea was this to be consistent with helm's cmd.go (https://github.com/operator-framework/operator-sdk/blob/master/internal/cmd/helm-operator/run/cmd.go). Since we are scaffolding the required code to set up the controller in hybrid operator, this package is no longer required there. Instead this would be used with helm operator.

Reply:
> Instead this would be used with helm operator.

As in "only in `helm-operator run`"?

If so, does my suggestion of putting these flags on an unexported struct in the CLI layer make sense or am I missing another piece of how this is expected to be used?

I would suggest not exposing these flags in the library part of the code base unless we expect users to use them (and even then, I might argue against them being user-facing).

Reply:
Agreed. Moved flags to internal/.